src/state/mod.rs

```diff
@@ -83,25 +187,73 @@ impl Default for Config {
 
 pub struct State {
     pub project_dirs: ProjectDirs,
-    pub config: ConfigWrapper<Config>,
-    pub profiles: ConfigWrapper<ModProfiles>,
+    pub config: ConfigWrapper<VersionAnnotatedConfig>,
+    pub mod_data: ConfigWrapper<VersionAnnotatedModData>,
     pub store: Arc<ModStore>,
 }
 
 impl State {
-    pub fn new() -> Result<Self> {
+    pub fn init() -> Result<Self> {
         let project_dirs = ProjectDirs::from("", "", "drg-mod-integration")
             .context("constructing project dirs")?;
         std::fs::create_dir_all(project_dirs.cache_dir())?;
         std::fs::create_dir_all(project_dirs.config_dir())?;
-        let config = ConfigWrapper::<Config>::new(project_dirs.config_dir().join("config.json"));
-        let profiles =
-            ConfigWrapper::<ModProfiles>::new(project_dirs.config_dir().join("profiles.json"));
+
+        let config_path = project_dirs.config_dir().join("config.json");
+
+        let config = if config_path.exists() {
+            let config = std::fs::read(&config_path);
+            let config = match config {
+                Err(_) => MaybeVersionedConfig::default(),
+                Ok(buf) => serde_json::from_slice::<MaybeVersionedConfig>(&buf)
+                    .context("failed to deserialize user config into maybe versioned config")?,
+            };
```

Comment:
Instead of checking `exists` and then `read` I would just attempt to `read` and check the error for `kind() == NotFound`:
```rust
    match std::fs::read(&config_path) {
        Ok(config) => Ok(...),
        Err(e) if e.kind() == ErrorKind::NotFound => Ok(Default::default()),
        Err(e) => Err(e),
    }?
```

Reply:
Changed all three checks (config, profiles/mod_data, cache) to attempt to read instead of checking for exists.